core/src/utils/mod.rs

```diff
@@ -100,3 +100,20 @@ pub fn new_array_vec<T: Debug, const N: usize>(capacity: usize) -> [Vec<T>; N] {
         .try_into()
         .expect("failed to convert vector to array")
 }
+
+#[test]
+#[should_panic]
+fn debug_assert_is_checked_in_ci() {
+    use std::env;
+
+    match env::var("CI").ok() {
+        // enforce the CI checks will always run `debug_assert`.
+        //
+        // `release` mode will skip these checks. however, it is desirable to have these checks not
+        // only because we perform a couple of them in miden, but also because they are included
+        // upstream.
+        Some(flag) if flag == "true" => debug_assert!(true),
```

Comment:
Should it be `debug_assert!(false),` here? This way, in CI mode, `debug_assert` will fail, and thus, the test will pass. Or am I missing something?

Also, I wonder if we should simplify this to be something like:
```Rust
#[test]
#[should_panic]
fn debug_assert_is_checked() {
    debug_assert!(false);
}
```
Basically, whenever we run tests (whether in CI or not) we always want to make sure that `debug_assert`'s are executed.

Reply:
> That is correct, it should be `false`. This commit [a75766d](https://github.com/maticnetwork/miden/commit/a75766df2d5d6a247e231e1556921bc37435e2b2) was added just as evidence for [#433 (comment)](https://github.com/maticnetwork/miden/issues/433#issuecomment-1282798312) and should be removed before we merge. I'll revert now.

Got it! Thank you!

> I thought also of adding this test as `ignore`, and in the CI we run the tests including the ignored ones. That way, we allow the user to `$ cargo test --release`, but it would fail if he `$ cargo test --release -- --include-ignored`. wdyt?

I think that we should try to make CI and non-CI runs as consistent as possible. Otherwise, a user could run tests locally (e.g., using `$ cargo test --release`) and everything would pass. But once they push their commits, some tests in CI could fail. This may lead to some confusion.

So, my vote would be to force the user to use the flags even locally and not make a CI vs. non-CI distinction.

Reply:
That makes sense also because a successful result of a local `$ cargo test --release` might be misleading in the sense of passing locally when some `debug_assert` somewhere is failing. Gotcha, updating!